### PR TITLE
fix fixable flaws in web/api/webrtc_api/

### PR DESCRIPTION
**Pull request by peterbe**
Note, I had to use the https://github.com/mdn/yari/pull/2660 branch to be able get this to work. 
The command I used, in `yari`, was:
```
BUILD_FIX_FLAWS=true BUILD_FOLDERSEARCH=web/api/webrtc_api yarn build
```

### CODE REVIEW

**Comment by peterbe** [comment]
@nschonni if you have a minute can you check too?

File: files/en-us/web/api/webrtc_api/connectivity/index.html [hunk 1]
```diff
@@ -87,7 +87,7 @@ <h3 id="UDP_candidate_types">UDP candidate types</h3>
 
 <dl>
  <dt><code>host</code></dt>
- <dd>A host candidate is one for which its {{domxref("RTCIceCandidate.ip", "ip")}} address is the actual, direct IP address of the remote peer.</dd>
+ <dd>A host candidate is one for which its {{domxref("RTCIceCandidate/address", "ip")}} address is the actual, direct IP address of the remote peer.</dd>
```
Comment (jpmedley) [inline files/en-us/web/api/webrtc_api/connectivity/index.html:90]:
```suggestion
 <dd>A host candidate is one for which its IP {{domxref("RTCIceCandidate.address", "address")}} is the actual, direct IP address of the remote peer.</dd>
```

File: files/en-us/web/api/webrtc_api/signaling_and_video_calling/index.html [hunk 2]
```diff
@@ -276,7 +276,7 @@ <h4 id="Starting_a_call">Starting a call</h4>
...
 
-<p>Once the <code>RTCPeerConnection</code> has been created, we request access to the user's camera and microphone by calling {{domxref("MediaDevices.getUserMedia()")}}, which is exposed to us through the {{domxref("Navigator.mediaDevices.getUserMedia")}} property. When this succeeds, fulfilling the returned promise, our <code>then</code> handler is executed. It receives, as input, a {{domxref("MediaStream")}} object representing the stream with audio from the user's microphone and video from their webcam.</p>
+<p>Once the <code>RTCPeerConnection</code> has been created, we request access to the user's camera and microphone by calling {{domxref("MediaDevices.getUserMedia()")}}, which is exposed to us through the {{domxref("MediaDevices/getUserMedia")}} property. When this succeeds, fulfilling the returned promise, our <code>then</code> handler is executed. It receives, as input, a {{domxref("MediaStream")}} object representing the stream with audio from the user's microphone and video from their webcam.</p>
```
Comment (jpmedley) [inline files/en-us/web/api/webrtc_api/signaling_and_video_calling/index.html:279]:
```suggestion
<p>Once the <code>RTCPeerConnection</code> has been created, we request access to the user's camera and microphone by calling {{domxref("MediaDevices.getUserMedia()")}}, which is exposed to us through the {{domxref("MediaDevices/getUserMedia")}} method. When this succeeds, fulfilling the returned promise, our <code>then</code> handler is executed. It receives, as input, a {{domxref("MediaStream")}} object representing the stream with audio from the user's microphone and video from their webcam.</p>
```

File: files/en-us/web/api/webrtc_api/taking_still_photos/index.html [hunk 3]
```diff
@@ -218,14 +218,14 @@ <h2 id="Fun_with_filters">Fun with filters</h2>
...
 
-<p>You can, if needed, restrict the set of permitted video sources to a specific device or set of devices. To do so, call {{domxref("navigator.mediaDevices.enumerateDevices()")}}. When the promise is fulfilled with an array of {{domxref("MediaDeviceInfo")}}  objects describing the available devices, find the ones that you want to allow and specify the corresponding {{domxref("MediaTrackConstraints.deviceId", "deviceId")}} or <code>deviceId</code>s in the {{domxref("MediaTrackConstraints")}} object passed into {{domxref("MediaDevices.getUserMedia", "getUserMedia()")}}.</p>
+<p>You can, if needed, restrict the set of permitted video sources to a specific device or set of devices. To do so, call {{domxref("MediaDevices/enumerateDevices")}}. When the promise is fulfilled with an array of {{domxref("MediaDeviceInfo")}}  objects describing the available devices, find the ones that you want to allow and specify the corresponding {{domxref("MediaTrackConstraints.deviceId", "deviceId")}} or <code>deviceId</code>s in the {{domxref("MediaTrackConstraints")}} object passed into {{domxref("MediaDevices.getUserMedia", "getUserMedia()")}}.</p>
```
Comment (jpmedley) [inline files/en-us/web/api/webrtc_api/taking_still_photos/index.html:221]:
```suggestion
<p>You can, if needed, restrict the set of permitted video sources to a specific device or set of devices. To do so, call {{domxref("MediaDevices.enumerateDevices")}}. When the promise is fulfilled with an array of {{domxref("MediaDeviceInfo")}}  objects describing the available devices, find the ones that you want to allow and specify the corresponding {{domxref("MediaTrackConstraints.deviceId", "deviceId")}} or <code>deviceId</code>s in the {{domxref("MediaTrackConstraints")}} object passed into {{domxref("MediaDevices.getUserMedia", "getUserMedia()")}}.</p>
```

File: files/en-us/web/api/webrtc_api/taking_still_photos/index.html [hunk 4]
```diff
@@ -218,14 +218,14 @@ <h2 id="Fun_with_filters">Fun with filters</h2>
...
 
 <h2 id="See_also">See also</h2>
 
 <ul>
  <li><a href="https://mdn-samples.mozilla.org/s/webrtc-capturestill">Try this sample</a></li>
  <li><a href="https://github.com/mdn/samples-server/tree/master/s/webrtc-capturestill">Sample code on Github</a></li>
- <li>{{domxref("Navigator.mediaDevices.getUserMedia()")}}</li>
+ <li>{{domxref("MediaDevices/getUserMedia")}}</li>
```
Comment (jpmedley) [inline files/en-us/web/api/webrtc_api/taking_still_photos/index.html:228]:
```suggestion
 <li>{{domxref("MediaDevices.getUserMedia")}}</li>
```

**Review by jpmedley — CHANGES_REQUESTED** [review]

File: files/en-us/web/api/webrtc_api/build_a_phone_with_peerjs/build_the_server/index.html [hunk 5]
```diff
@@ -49,6 +49,6 @@
 
 <p>You should be able to connect to your app via local host, in my <code>server.js</code> I’m using port 8000(defined on line 7) but you may be using another port number. Run <code>node .</code> in your terminal and visit <code>localhost:8000</code> in your browser and you should see a page that looks like this:</p>
 
-<p><img alt="A cream background with the words 'phone a friend' in bold, dark green font as the heading. 'Connecting...' is immediately below that and 'please use headphones!' below that. Following on, a big dark green button with 'Call' written in the same cream color of the background. " src="https://miro.medium.com/max/1400/1*IC3eVvGnzvG1koQuxO0MMw.png" style="height: 894px; width: 1400px;"></p>
+<p><img alt="A cream background with the words 'phone a friend' in bold, dark green font as the heading. 'Connecting...' is immediately below that and 'please use headphones!' below that. Following on, a big dark green button with 'Call' written in the same cream color of the background. " src="1ic3evvgnzvg1koquxo0mmw.png"></p>
```
Comment (nschonni) [inline files/en-us/web/api/webrtc_api/build_a_phone_with_peerjs/build_the_server/index.html:52]: Yup, looks like the "bad" characters are striped off now. Not sure if replacing them vs. removing entirely, but all these ones are all junk names anyway

**Review by nschonni — APPROVED** [review]

**Comment by peterbe** [comment]
So, unfortunately, we're still suffering from https://github.com/mdn/yari/issues/2520
The code that fixes the value in `{{domxref(...THIS...)}}` isn't working as it should. 
But as pointed out here: https://github.com/mdn/yari/pull/2551#pullrequestreview-569561937 we need to treat `domxref` differently from other xref macros. Annoying. 

@jpmedley I manually fixed the places where it should have been a `.` instead of the `/`. And then I just directly committed your suggestions.

**Comment by peterbe** [comment]
@jpmedley ping r?

**Comment by peterbe** [comment]
@jpmedley ping

File: files/en-us/web/api/webrtc_api/connectivity/index.html [hunk 6]
```diff
@@ -87,7 +87,7 @@ <h3 id="UDP_candidate_types">UDP candidate types</h3>
 
 <dl>
  <dt><code>host</code></dt>
- <dd>A host candidate is one for which its {{domxref("RTCIceCandidate.ip", "ip")}} address is the actual, direct IP address of the remote peer.</dd>
+ <dd>A host candidate is one for which its {{domxref("RTCIceCandidate.address", "ip")}} address is the actual, direct IP address of the remote peer.</dd>
```
Comment (jpmedley) [inline files/en-us/web/api/webrtc_api/connectivity/index.html:90]:
```suggestion
 <dd>A host candidate is one for which its {{domxref("RTCIceCandidate.address", "ip address")}} is the actual, direct IP address of the remote peer.</dd>
```

**Review by jpmedley — CHANGES_REQUESTED** [review]

**Comment by peterbe** [comment]
This now too rotten and too many merge conflicts. I looked at a couple of pages and the diffs were way too difficult for me to spend the time on.

**Pull request closed without merging**